### DISCUSSION TITLE
I cannot open 'spotube' on mac because Apple cannot check it for malicious software

### DISCUSSION THREAD

**Discussion by kanodiaayush** [KRTirtho/spotube#1097] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

![image](https://github.com/KRTirtho/spotube/assets/1688433/11d2ee9b-fb6d-4870-aeeb-f445f5fdef2b)

### Expected Behavior

Should run out of the box.

### Steps to reproduce

1. I just opened the app using cmd + space on mac

### Operating System

mac OS, M1, Venture 13.5

### Installation source

GitHub Releases (Binary)

**Comment by verhovsky** [comment, marked as answer]
Go to System Settings > Privacy & Security > Security (scroll down) > Open Anyway > Open

<img width="715" alt="Screenshot 2024-01-20 at 22 37 34" src="https://github.com/KRTirtho/spotube/assets/5687998/c2eeff81-fb96-47d4-ad16-c04c5945299f">

**Comment by kanodiaayush** [comment]
Do I have something to worry about? Is it just safe to do this?

**Comment by verhovsky** [comment]
I think this means they haven't [notarized](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution) the app. Overall, you shouldn't worry too much if you trust the developers of Spotube. You could [build Spotube from source](https://github.com/KRTirtho/spotube/blob/master/CONTRIBUTION.md#your-first-code-contribution), but you'll probably still have to do the "Open Anyway" thing with the resulting binary.

**Comment by HappyGoFishing** [comment]
This is apple's attempt at nudging you into the walled garden.

**Comment by KRTirtho** [comment]
Converting it to discussion as it's a question rather than a bug

**Comment by kanodiaayush** [comment]
Thanks. I'll request the maintainers of Spotube to considering go through notarization if its not too hard, since it then lets me use this without any fear (and perhaps others).

**Reply by verhovsky** [reply]
@KRTirtho it would be nice if you could notarize the app so that people don't have to go through this tedious step and not everyone knows about it. Here is how VS Code enabled notarization with azure pipelines https://github.com/microsoft/vscode/pull/86588/files

From a quick google search, there seems to be a github action to do this https://github.com/marketplace/actions/xcode-notarization

https://docs.flutter.dev/platform-integration/macos/building#building-macos-apps

**Comment by Trapper4888** [comment]
@KRTirtho Is there a reason you can't or don't want to do the steps mentioned by @verhovsky ?

@verhovsky If you right-click the app in the finder and click open, you also have the option to open anyway, without going through the settings.

**Comment by micalx** [comment]
Hello @Trapper4888 and @verhovsky 
both of your workarounds don't work for me on macOS 15.3. Any ideas?

@KRTirtho I would kindly ask you to think about the norarisation again. Maye we can assist you?

Thank you & best regards
Michael

**Comment by zemua** [comment]
I cannot "open anyway" because my macbook is managed by my company and they don't allow that.
I did have it on the previous one, but they seem to have gotten stricter.

Since the app is built in flutter, maybe you could include the web build in the release so it can be run from the browser?
